controller/src/main/java/io/pravega/controller/server/rpc/grpc/impl/GRPCServerConfigImpl.java

```diff
@@ -73,6 +78,7 @@ public GRPCServerConfigImpl(final int port, final String publishedRPCHost, final
         this.authorizationEnabled = authorizationEnabled;
         this.userPasswordFile = userPasswordFile;
         this.tlsEnabled = tlsEnabled;
+        this.tlsProtocolVersion = tlsProtocolVersion;
```

Comment:
The reason I asked you to use an enum is not just for validation. It's also so that you may convert your config string from whatever format you have in the config file into something that's easy for you to use in the code. I see you only invoke `parse` for validation purposes, but you still carry your full string everywhere and then use `.split(',')` where needed. 

I suggest you either parse the config value here and hold it into an array (which you then pass anyway to the `GrpcSslContexts`, or you create a class that takes in your string, parses it (and validates it in the meantime), and holds it in some form internally. It can then either return it as a string (i.e., in `toString`) or as an array, depending on how you use it.

Reply:
@kotlasaicharanreddy , I think what Andrei is talking about is something like this:

```
import java.util.Arrays;

public class TLSProtocolVersion {
    private String[] tlsProtocols;

    public TLSProtocolVersion(String s) {
        tlsProtocols = TlsProtocolVersion.parse(s);
    }

    public String[] tlsProtocols() {
        return tlsProtocols;
    }

    public enum TlsProtocolVersion {
        TLSv1_2("TLSv1.2"),
        TLSv1_3("TLSv1.3"),
        TLS1_2ANDTLS1_3("TLSv1.2,TLSv1.3"),
        TLS1_3ANDTLS1_2("TLSv1.3,TLSv1.2");

        String protocol;

        TlsProtocolVersion(String tlsprotocol) {
            protocol = tlsprotocol;
        }

        public static String[] parse(String s) {
            if (Arrays.stream(TlsProtocolVersion.values()).anyMatch(e -> e.protocol.equals(s))) {
                return s.split(",");
            }
            return null;
        }
    }
}
```

With that (really just a slightly re-arranged version of your code where all we're doing is only parsing once and holding on the result which can be looked at for validation and used as is when needed.

We validate it and anywhere we need this property, it's available in its parsed -ready-to-use form. (String array)

```
TLSProtocolVersion tlsp = new TLSProtocolVersion(tlsProtocolVersion);  
Preconditions.checkArgument( tlsp.tlsProtocols() != null, "message...");

tlsp.tlsProtocols(); holds the String[] that can be passed directly, already parsed, into the TLSHelpers. etc.
       
```

Reply:
Modified `TLSProtocolVersion` class as suggested. Thanks.